### PR TITLE
Upgrading Kotlin-Native compiler rules to 1.4.0

### PR DESCRIPTION
**Pull request by alxrsngrtn**
Upgraded our Kotlin-Native compiler rules (used for Wasm compilation) to 1.4.0. Updated documentation of the `repo.bzl` file to describe how others can update the compiler in the future. 

CC: @mykmartin @bgogul

### CODE REVIEW

**Review by cromwellian — COMMENTED** [review]
Does the EMSDK also need to be bumped?

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 1]
```diff
@@ -343,23 +353,16 @@ def arcs_kt_particles(
...
-        wasm_particle_lib = name + "-lib" + _WASM_SUFFIX
-        arcs_kt_native_library(
-            name = wasm_particle_lib,
-            srcs = wasm_srcs,
-            deps = wasm_deps,
-        )
-
         # Create a kt_native_binary that groups everything together.
         native_binary_name = name + _WASM_SUFFIX
         kt_native_binary(
```
Comment (cromwellian) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl:358]: should you be using arcs_kt_native_library since you added it above?

File: third_party/java/arcs/build_defs/internal/kotlin.bzl [hunk 2]
```diff
@@ -160,6 +160,16 @@ def arcs_kt_native_library(**kwargs):
     kwargs["kotlincopts"] = merge_lists(kotlincopts, COMMON_KOTLINC_OPTS)
     kt_native_library(**kwargs)
 
+def arcs_kt_native_binary(**kwargs):
```
Comment (cromwellian) [inline third_party/java/arcs/build_defs/internal/kotlin.bzl:163]: Why is this added here, but deleted below?

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 1 above)
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:358]: Yes, I should be using it, thank you.

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 2 above)
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:163]: This was added here because the kt_native rules require that the "final" wasm artifact be a kt_native_binary. This macro adds kotlincopts to the regular native_binary invokation. 
Below, the deleted portion was a native_library macro.

**Review by cromwellian — APPROVED** [review]
Approved

Is the latest @emsdk always pulled or is it pinned?

**Comment by alxrsngrtn** [comment]
> Approved
> 
> Is the latest @emsdk always pulled or is it pinned?

These rules are independent of the emsdk. I think they could use an upgrade, and we can consider that in a separate PR.

**Comment by alxrsngrtn** [comment]
Presubmit can be found here: cl/330024611

**Review by csilvestrini — APPROVED** [review]

**Comment by alxrsngrtn** [comment]
I'm going to meet w/ @mpetrov next week to finalize this PR.

File: third_party/java/arcs/build_defs/internal/kotlin.bzl (hunk 2 above)
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/kotlin.bzl:163]: After talking to Martin, this change is no longer relevant.
